### PR TITLE
Switch helloworld-go with kn

### PR DESCRIPTION
**Pull request by mattmoor**
WIP as this is based on https://github.com/knative/docs/pull/2805

### CODE REVIEW

**Comment by mattmoor** [comment]
@abrennan89 @rhuss I did a similar thing in this PR with `helloworld-go`, check out this section:
https://docs-2807.default.docs-on-the-rocks.io/docs/serving/samples/hello-world/helloworld-go/#building-and-deploying-the-sample

**Comment by mattmoor** [comment]
Ok, this sample is RFAL here: https://docs-2807.default.docs-on-the-rocks.io/docs/serving/samples/hello-world/helloworld-go/#building-and-deploying-the-sample

I updated the tab default to `kn` here as well

**Comment by mattmoor** [comment]
I created https://docs.google.com/spreadsheets/d/1Nbp1nb0V1l-8ZfdGnXZMfrcgt6POS9ZbT94Sa8GGFhc/edit#gid=0 to crowd-source going wide on these samples once we have a canonical one landed.

File: docs/serving/samples/hello-world/helloworld-go/index.md
Comment (abrennan89) [inline docs/serving/samples/hello-world/helloworld-go/index.md:163]:
```suggestion
   Check that the container image value in the `service.yaml` file matches the container you built in the previous step.
   
   Apply the configuration using `kubectl`:

   ```shell
   kubectl apply --filename service.yaml
   ```
```

Comment (abrennan89) [reply docs/serving/samples/hello-world/helloworld-go/index.md:163]: No. 3 would be "apply..."

File: docs/serving/samples/hello-world/helloworld-go/index.md
Comment (abrennan89) [inline docs/serving/samples/hello-world/helloworld-go/index.md:165]:
```suggestion
   After your service is created, Knative will perform the following steps:
```

File: docs/serving/samples/hello-world/helloworld-go/index.md
Comment (abrennan89) [inline docs/serving/samples/hello-world/helloworld-go/index.md:183]: No. 4; also I feel like this should be a verification step instead.
I'm not sure that tabs are the best way to display this info at all tbh...

File: docs/serving/samples/hello-world/helloworld-go/index.md
Comment (abrennan89) [inline docs/serving/samples/hello-world/helloworld-go/index.md:188]:
```suggestion
   Use `kn` to deploy the service:
```

File: docs/serving/samples/hello-world/helloworld-go/index.md
Comment (abrennan89) [inline docs/serving/samples/hello-world/helloworld-go/index.md:194]: would be much better to just show an expected example output, e.g, command for how they get the URL and a code block that shows a sample.

File: docs/serving/samples/hello-world/helloworld-go/index.md
Comment (abrennan89) [inline docs/serving/samples/hello-world/helloworld-go/index.md:200]: This should have a new heading and should be a section containing the verification steps, IMHO

**Review by abrennan89 — COMMENTED** [review]
@mattmoor left a bunch of comments.
In terms of doc structure, I'm really not sure everything in one big doc like this with tabs for samples is the way to go.
I think it's less readable than just creating one focused procedure for each, e.g.
- "Using the `helloworld-go` sample app with `kn` CLI"
- "Using the `helloworld-go` sample app with YAML"

that live in the same page under the heading of `helloworld-go` sample.
Even the prerequisites doc'd this way aren't really correct because they don't mention installing `kn` and that is a pre-req but only if you use the procedure using `kn`, so I think they should be separate.

File: docs/serving/samples/hello-world/helloworld-go/index.md
Comment (rhuss) [reply docs/serving/samples/hello-world/helloworld-go/index.md:21]: +1 for the same document structure across all examples. The suggestion looks good to me.

**Review by rhuss — COMMENTED** [review]
Looks good, but I would like to add the following suggestions:

* Kn should be added to the prerequisites.
* The command to show the URL can be also performed with `kn describe service helloworld-go -o url`
* The service can be deleted with `kn service delete helloworld-go`

The latter two commands can be also added as tabs. I'm just preparing an example for the shell sample, which could be used here as blueprint, too.

**Comment by rhuss** [comment]
I really wonder why the test is passing as README.md is removed but `sampleconsistency` test explicitly tries to read it, so that the e2e test should fail here.

/test pull-knative-docs-unit-tests

**Comment by rhuss** [comment]
ah, here we go ;-) test fails now, too. I'm going to fix this issue / check along with https://github.com/knative/docs/pull/2832

File: docs/serving/samples/hello-world/helloworld-go/index.md
Comment (mattmoor) [reply docs/serving/samples/hello-world/helloworld-go/index.md:21]: I followed the headings from @rhuss PR, which I now realize are difference from here.  +1 for consistency, I just need to know what we want the shape to be 😅

File: docs/serving/samples/hello-world/helloworld-go/index.md
Comment (mattmoor) [reply docs/serving/samples/hello-world/helloworld-go/index.md]: Added sample output.

File: docs/serving/samples/hello-world/helloworld-go/index.md
Comment (mattmoor) [reply docs/serving/samples/hello-world/helloworld-go/index.md:210]: Added `## Verifying` above this.

File: docs/serving/samples/hello-world/helloworld-go/index.md
Comment (mattmoor) [reply docs/serving/samples/hello-world/helloworld-go/index.md]: What do you suggest?  We'd been talking about tabs for a while, which is why I'd bit the bullet here.  Definitely open to suggestions.

File: docs/serving/samples/hello-world/helloworld-go/index.md
Comment (mattmoor) [reply docs/serving/samples/hello-world/helloworld-go/index.md:118]: Ack, we probably need to update the other samples.

**Review by mattmoor — COMMENTED** [review]
ok, I think I got all the feedback incorporated.  A lot of this was carryover from README.md -> index.md, so we may want to do a pass over the other languages to ensure we're consistently applying the feedback to all of them.

I just pushed a new commit with the changes, and I've rebased this on https://github.com/knative/docs/pull/2832 where @rhuss fixed the test.  I'll check the staged site shortly and make sure things look alright.

**Comment by mattmoor** [comment]
Needs https://github.com/knative/docs/pull/3004 to fix the local build, but when I pulled it up it looked fine (after some tweaks).

File: docs/serving/samples/hello-world/helloworld-go/index.md
Comment (abrennan89) [inline docs/serving/samples/hello-world/helloworld-go/index.md:18]:
```suggestion
- Optional: You can use the Knative CLI client [`kn`](https://github.com/knative/client/releases) to simplify resource creation and deployment. Alternatively, you can use `kubectl` to apply resource files directly.
```

File: docs/serving/samples/hello-world/helloworld-go/index.md
Comment (abrennan89) [inline docs/serving/samples/hello-world/helloworld-go/index.md:20]:
```suggestion
## Building
```

We use deploying further down so I'd make them all gerunds or none of them

**Review by abrennan89 — COMMENTED** [review]
Minor suggestions, otherwise lgtm :)

File: docs/serving/samples/hello-world/helloworld-go/index.md
Comment (mattmoor) [reply docs/serving/samples/hello-world/helloworld-go/index.md]: Fixed here and in the shell sample I copied it from.  thanks!

File: docs/serving/samples/hello-world/helloworld-go/index.md
Comment (mattmoor) [reply docs/serving/samples/hello-world/helloworld-go/index.md]: Same, fixed both.

**Review by abrennan89 — APPROVED** [review]
/hold
/lgtm

Thanks Matt! I'm fine with merging this now :slightly_smiling_face: sticking a hold on it to make sure you're done with commits first, but feel free to remove

**Comment by mattmoor** [comment]
/unhold

Yep, done for now.  Thanks, @abrennan89 and sorry this took so long for me to circle back 😅

**Comment by abrennan89** [comment]
> Yep, done for now. Thanks, @abrennan89 and sorry this took so long for me to circle back sweat_smile

lol no problem at all, thank you!

**Pull request merged**
